NIFI-8677 Add custom endpoint suffix for Azure EventHub Processors

timeabarna opened the pull request:
https://issues.apache.org/jira/browse/NIFI-8677

#### Description of PR

Enables to use special regions including US Government, China, and Germany for Azure EventHub processors.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [x] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

timeabarna commented:
@jfrazee Hello Joey, can you please check this PR when you have some time? Thanks in advance.

timeabarna commented:
@jfrazee Hello Joey, can you please check this PR when you have some time? Thanks in advance.

jfrazee commented:
@timeabarna Sorry for taking so long. I'm starting to look at this finally.

timeabarna commented:
Thanks @jfrazee for your help. I've modified the code based on your recommendation.

jfrazee reviewed (changes requested):
@timeabarna Hey, I missed something about how we've changed the endpoint names to remove the leading `.`

That's going to create an upgrade issue for `GetAzureEventHub`. Wanted to see what your thoughts were on whether we revert back to using the leading `.` or live with having to manually intervene for an upgrade.

timeabarna commented:
Hello @jfrazee, I've added back the leading '.' to servicebus path, and modified Consume and Put processor code accordingly. Thanks a lot for your help.

timeabarna commented on nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/utils/AzureEventHubUtils.java:
```diff
@@ -33,6 +34,10 @@
+    public static final AllowableValue AZURE_ENDPOINT = new AllowableValue("servicebus.windows.net","Azure", "Servicebus endpoint for general use");
+    public static final AllowableValue AZURE_CHINA_ENDPOINT = new AllowableValue("servicebus.chinacloudapi.cn", "Azure China", "Servicebus endpoint for China");
+    public static final AllowableValue AZURE_GERMANY_ENDPOINT = new AllowableValue("servicebus.cloudapi.de", "Azure Germany", "Servicebus endpoint for Germany");
+    public static final AllowableValue AZURE_US_GOV_ENDPOINT = new AllowableValue("servicebus.usgovcloudapi.net", "Azure US Government", "Servicebus endpoint for US Government");
```
I agree with you @jfrazee that option 1 is the better solution, modified the code based on that alternative.

pvillard31 commented:
@jfrazee @turcsanyip - what do you think about the current state of this pull request?
